requests_auth/authentication.py

```diff
@@ -273,15 +299,15 @@ def __call__(self, r):
 
     def request_new_token(self) -> tuple:
         # As described in https://tools.ietf.org/html/rfc6749#section-4.4.3
-        token, expires_in = request_new_grant_with_post(
+        token, expires_in, _refresh_token = request_new_grant_with_post(
             self.token_url,
             self.data,
             self.token_field_name,
             self.timeout,
             self.session,
         )
         # Handle both Access and Bearer tokens
-        return (self.state, token, expires_in) if expires_in else (self.state, token)
+        return (self.state, token, expires_in, _refresh_token) if expires_in else (self.state, token)
```

Comment:
If we don't expect a refresh token and as it is optional, we can skip it?
```suggestion
        return (self.state, token, expires_in) if expires_in else (self.state, token)
```

Reply:
Yes this is possible. Changed this and added an extra case in the `TokenMemoryCache.getToken()` method for when the length of the tuple is 3.